Fabric.Authorization.API/Services/IdentitySearchService.cs

```diff
@@ -1,23 +1,110 @@
-﻿using System.Threading.Tasks;
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 using Fabric.Authorization.API.Models.Search;
+using Fabric.Authorization.API.Services.External.Identity;
+using Fabric.Authorization.Domain.Models;
 using Fabric.Authorization.Domain.Stores.Services;
+using Nancy.Extensions;
 
 namespace Fabric.Authorization.API.Services
 {
     public class IdentitySearchService
     {
         private readonly ClientService _clientService;
+        private readonly GroupService _groupService;
+        private readonly IIdentityServiceProvider _identityServiceProvider;
+        private readonly RoleService _roleService;
 
-        public IdentitySearchService(ClientService clientService)
+        public IdentitySearchService(
+            ClientService clientService,
+            RoleService roleService,
+            GroupService groupService,
+            IIdentityServiceProvider identityServiceProvider)
         {
             _clientService = clientService;
+            _roleService = roleService;
+            _groupService = groupService;
+            _identityServiceProvider = identityServiceProvider;
         }
 
-        public async Task<IdentitySearchResponse> Search(IdentitySearchRequest request)
+        public async Task<IEnumerable<IdentitySearchResponse>> Search(IdentitySearchRequest request)
         {
+            var searchResults = new List<IdentitySearchResponse>();
             var client = await _clientService.GetClient(request.ClientId);
+            var roles = await _roleService.GetRoles();
+            var clientRoles = new List<Role>();
 
-            return new IdentitySearchResponse();
+            foreach (var role in roles)
```

Comment:
Should we encapsulate this code in a function in the RoleService? Seems like we might want to query all roles for a client for another reason at some point.